sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample12_ConsumeEventsWithEventProcessor.cs

```diff
@@ -35,17 +40,250 @@ public class Sample12_ConsumeEventsWithEventProcessor : IEventHubsSample
         public async Task RunAsync(string connectionString,
                                    string eventHubName)
         {
-            // We will start by creating a client using its default set of options.
+            // We will start by creating a client using its default set of options.  It will be used by the event processor to
+            // communicate with the Azure Event Hubs service.
 
             await using (var client = new EventHubClient(connectionString, eventHubName))
             {
-                //TODO (caio, pri1): Populate the sample
+                // An event processor is associated with a specific Event Hub and a consumer group.  It receives events from
+                // all existing partitions in the Event Hub, passing them to the user for processing.
+                //
+                // Partition processors are associated with a specific partition and are responsible for processing events
+                // received from the event processor.  Their implementation must be provided by the user by creating a class
+                // that implements the IPartitionProcessor interface.
+                //
+                // An event processor must be able to create a partition processor for every partition it is responsible for.
+                // For this reason, a Func-typed factory must be passed to its constructor.  The factory returns a partition
+                // processor created from two arguments:
+                //
+                //   A partition context: contains information about the partition the partition processor will be processing
+                //   events from.  In this sample, we are only interested in its partition id.
+                //
+                //   A checkpoint manager: responsible for the creation of checkpoints.  It's not used in this sample.
+                //
+                // We'll be using a SamplePartitionProcessor, whose implementation can be found at the end of this sample.  Its
+                // constructor takes the associated partition id so it can provide useful log messages.
+
+                Func<PartitionContext, CheckpointManager, IPartitionProcessor> partitionProcessorFactory =
+                    (partitionContext, checkpointManager) => new SamplePartitionProcessor(partitionContext.PartitionId);
+
+                // A partition manager is able to create checkpoints and list/claim partition ownership.  The user can implement
+                // their own partition manager by creating a subclass from the PartitionManager abstract class.  Here we are creating
+                // a new instance of an InMemoryPartitionManager, provided by the Azure.Messaging.EventHubs.Processor namespace.
+                // This won't be relevant to this sample, but it must be passed to the event processor constructor.
```

Comment:
Maybe add just a bit of context with "... This isn't relevant to understanding this sample, but is required by.... "?